internal/app/machined/pkg/controllers/siderolink/config.go

```diff
@@ -70,9 +73,17 @@ func (ctrl *ConfigController) Run(ctx context.Context, r controller.Runtime, _ *
 
 		r.StartTrackingOutputs()
 
-		if endpoint := ctrl.apiEndpoint(cfg); endpoint != "" {
+		if ep := ctrl.apiEndpoint(cfg); ep != "" {
+			var parsed endpoint.Endpoint
+
+			parsed, err = endpoint.Parse(ep)
+			if err != nil {
+				return fmt.Errorf("failed to parse siderolink API endpoint: %w", err)
+			}
+
 			if err = safe.WriterModify(ctx, r, siderolink.NewConfig(config.NamespaceName, siderolink.ConfigID), func(c *siderolink.Config) error {
-				c.TypedSpec().APIEndpoint = endpoint
+				c.TypedSpec().APIEndpoint = ep
```

Comment:
there's a bit of lack of symmetry now here.

the API endpoint might contain a join token and a grpc_tunnel flag, we parse the tunnel flag, but not the API token

we should either parse both here or none

Reply:
Fixed!